src/pineko/kfactor.py

```diff
@@ -52,145 +48,133 @@ def read_kfactor(kfactor_path):
 
 def construct_scales_array(
     mu2_ren_grid,
-    m_value,
     order,
-    new_order,
+    order_to_update,
     central_k_factor,
     bin_index,
     alphas,
-    order_exists,
 ):
     """Construct the array that will rescale the subgrid array taking into account the different renormalization scales."""
     scales_array = []
     for mu2 in mu2_ren_grid:
         scales_array.append(
             compute_scale_factor(
-                m_value,
                 order,
-                new_order,
+                order_to_update,
                 mu2,
                 central_k_factor,
                 bin_index,
                 alphas,
-                order_exists,
             )
         )
     return scales_array
 
 
 def compute_scale_factor(
-    m,
-    nec_order,
-    to_construct_order,
-    Q2,
+    order,
+    order_to_update,
+    mu2,
     central_k_factor,
     bin_index,
     alphas,
-    order_exists,
 ):
     """Compute the factor to be multiplied to the given nec_order.
 
     Parameters
     ----------
-    m : int
-        first non zero perturbative order
     nec_order : tuple(int)
         tuple of the order that has to be rescaled to get the final order
-    to_contruct_order : tuple(int)
-        tuple of the scale varied order to be constructed
-    Q2: float
+    order_to_update : tuple(int)
+        order to update
+    mu2: float
         energy scale squared of the bin
     central_k_factor: list(float)
         list of the centrals k-factors
     bin_index: int
         index of the bin
     alphas: lhapdf.AlphaS
         alpha_s object
-
-    Returns
-    -------
-    float
-        full contribution factor
     """
-    max_as = to_construct_order[0] - m
-    alpha_val = alphas.alphasQ2(Q2)
-    alpha_term = 1.0 / pow(alpha_val, max_as - (nec_order[0] - m))
+    alpha_val = alphas.alphasQ2(mu2)
+    max_as = order_to_update[0]
+    as_order = order[0]
+    alpha_term = 1.0 / pow(alpha_val, max_as - as_order)
     k_term = central_k_factor[bin_index] - 1.0
-    if order_exists and (max_as - (nec_order[0] - m)) == 0:
-        k_term = central_k_factor[bin_index]
     return k_term * alpha_term
 
 
 def scale_subgrid(extracted_subgrid, scales_array):
-    """Rescales the array contained in the subgrid using scales_array and returns a new subgrid constructed with the scaled array."""
+    """Rescales the array contained in the subgrid using scales_array."""
     original_array = extracted_subgrid.to_array3()
     if len(original_array) != len(scales_array):
```

Comment:
shouldn't we check on `.shape` here?

Reply:
I should have not changed anything here....

Reply:
you mean, instead of using `len` in line 109?

Reply:
> you mean, instead of using `len` in line 109?

yes - and [as said below](https://github.com/NNPDF/pineko/pull/161#discussion_r1528606957) we should collapse this with just `np.multiply`